internal/stacks/stackruntime/internal/stackeval/for_each.go

```diff
@@ -239,6 +242,17 @@ func noForEachInstancesMap[T any](makeInst func(addrs.InstanceKey, instances.Rep
 	}
 }
 
+func unknownForEachInstancesMap[T any](makeInst func(addrs.InstanceKey, instances.RepetitionData) T) map[addrs.InstanceKey]T {
+	return map[addrs.InstanceKey]T{
+		addrs.WildcardKey: makeInst(addrs.WildcardKey, instances.RepetitionData{
```

Comment:
Can you use `instances.UnknownForEachRepetitionData` here? You can get the type from the `instancesMap` function based on the `maybeForEachVal` value.